app/models/customer.rb

```diff
@@ -1,3 +1,19 @@
 class Customer < ApplicationRecord
   has_many :invoices
+  has_many :transactions, through: :invoices
+
+  def self.top_customers(number)
+    Customer.joins(:transactions)
+    .where("result = ?", "0")
+    .group(:id)
+    .select("customers.*, count(transactions) as number_of_transactions")
+    .order("number_of_transactions" => :desc)
+    .limit(number)
+  end
+
+  def number_of_successful_transactions
```

Comment:
we could also think about defining a scope that finds successful transactions since we use that in multiple places

Reply:
that's a great suggestion, no need to enum successful/failed, scope would be better and easier to call on. Will make that change.